[Collatz Conjecture]: Approaches (#3308)
exercism/python: exercises/practice/collatz-conjecture/.approaches/ternary-operator/content.md

# Ternary operator

```python
def steps(number):
    if number <= 0:
        raise ValueError("Only positive integers are allowed")
    counter = 0
    while number != 1:
        number = number / 2 if number % 2 == 0 else number * 3 + 1
        counter += 1
    return counter
```

This approach starts with checking if the number is less than or equal to zero.

> If it is, then it raises a [`ValueError`][value-error].

Reviewer: ```suggestion
If it is, then a [`ValueError`][value-error] is raised.
```

> After that, we declare a counter variable and set it to zero.

Reviewer: ```suggestion
After that, a counter variable is assigned to zero.
```

Then we start a `while` loop that will run until the number is equal to one.
Meaning the loop won't run if the number is already one.

> Inside the loop we have a [ternary operator][ternary-operator].

Reviewer: ```suggestion
Inside the loop we have a [ternary operator][ternary-operator] or [conditional expression][conditional-expression].
```

> A ternary operator is a one-line `if` and `else` statement.

Reviewer: ```suggestion
A ternary operator/conditional expression can be viewed as a one-line `if/else` statement.
```

> That means that we can make the code more concise by using it.

Reviewer: ```suggestion
Using a one-line construct can make the code more concise.
```

> We declare that number is equal to the result of the ternary operator.

Reviewer: ```suggestion
We assign the number value to the result of the ternary operator.
```

> The ternary operator checks if the number is even.

Reviewer: ```suggestion
The ternary operator/conditional expression checks if the number is even.
```

If it is, then we divide it by two.

> If it isn't, then we multiply it by three and add one.

Reviewer: ```suggestion
If the number is not even, we multiply by three and add one.
```

> After that, we increment the counter by one.

Reviewer: ```suggestion
Then the counter is incremented by one.
```

> After the loop is done, we return the counter variable.

Reviewer: ```suggestion
When the loop completes, we return the counter value.
```
